scripts/build/builders/ti.py

```diff
@@ -0,0 +1,140 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from enum import Enum, auto
+
+from .gn import GnBuilder
+from typing import Optional
+
+
+class TIApp(Enum):
+    LOCK = auto()
+    PUMP = auto()
+    PUMP_CONTROLLER = auto()
+    ALL_CLUSTERS = auto()
+    ALL_CLUSTERS_MINIMAL = auto()
+    LIGHTING = auto()
+    SHELL = auto()
+
+    def ExampleName(self):
+        if self == TIApp.LOCK:
+            return 'lock-app'
+        elif self == TIApp.PUMP:
+            return 'pump-app'
+        elif self == TIApp.PUMP_CONTROLLER:
+            return 'pump-controller-app'
+        elif self == TIApp.ALL_CLUSTERS:
+            return 'all-clusters-app'
+        elif self == TIApp.ALL_CLUSTERS_MINIMAL:
+            return 'all-clusters-minimal-app'
+        elif self == TIApp.LIGHTING:
+            return 'lighting-app'
+        elif self == TIApp.SHELL:
+            return 'shell'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    def AppNamePrefix(self, board):
+        if self == TIApp.LOCK:
+            return f'chip-{board.BoardName()}-lock-example'
+        elif self == TIApp.PUMP:
+            return f'chip-{board.BoardName()}-pump-example'
+        elif self == TIApp.PUMP_CONTROLLER:
+            return f'chip-{board.BoardName()}-pump-controller-example'
+        elif self == TIApp.ALL_CLUSTERS:
+            return f'chip-{board.BoardName()}-all-clusters-example'
+        elif self == TIApp.ALL_CLUSTERS_MINIMAL:
+            return f'chip-{board.BoardName()}-all-clusters-minimal-example'
+        elif self == TIApp.LIGHTING:
+            return f'chip-{board.BoardName()}-lighting-example'
+        elif self == TIApp.SHELL:
+            return f'chip-{board.BoardName()}-shell-example'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    def BuildRoot(self, root, board):
+        return os.path.join(root, 'examples', self.ExampleName(), board.FamilyName())
+
+
+class TIBoard(Enum):
+    LP_CC2652R7 = auto()
+    LP_EM_CC1354P10_6 = auto()
+
+    def BoardName(self):
+        if self == TIBoard.LP_CC2652R7:
+            return 'LP_CC2652R7'
+        elif self == TIBoard.LP_EM_CC1354P10_6:
+            return 'LP_EM_CC1354P10_6'
+        else:
+            raise Exception('Unknown board type: %r' % self)
+
+    def FamilyName(self):
+        if self == TIBoard.LP_CC2652R7:
+            return 'cc13x2x7_26x2x7'
+        elif self == TIBoard.LP_EM_CC1354P10_6:
+            return 'cc13x4_26x4'
+        else:
+            raise Exception('Unknown board type: %r' % self)
+
+
+class TIBuilder(GnBuilder):
+
+    def __init__(self,
+                 root,
+                 runner,
+                 board=TIBoard.LP_CC2652R7,
+                 app: TIApp = TIApp.LOCK,
+                 openthread_ftd: Optional[bool] = None):
```

Comment:
should we add ftd as a modifier here and also noftd? Otherwise this does not seem used (based on content of `all_tagets_linux_x64.txt`

Reply:
Noticed now that you auto-expanded platforms. I think arguments should be modifiers unless we have strong reasons to split them out.